cli/client.go

```diff
@@ -380,6 +387,83 @@ func RobotsPartShellAction(c *cli.Context) error {
 	)
 }
 
+// checkUpdateResponse holds the values used to hold release information.
+type checkUpdateResponse struct {
+	Name       string `json:"name"`
+	TagName    string `json:"tag_name"`
+	TarballURL string `json:"tarball_url"`
+}
+
+// CheckUpdateAction is the corresponding Action for 'check-update'.
+func CheckUpdateAction(c *cli.Context) error {
+	if c.Bool(quietFlag) {
+		return nil
+	}
+
+	dateCompiledRaw := rconfig.DateCompiled
```

Comment:
Aside that just popped into my head: if this is empty, we should probably just return nil and assume that the user knows what they are doing.

People who install the cli with `go install go.viam.com/rdk/cli` (or who build it locally with go build) won't have this set and this could get annoying if we print `warningf(c.App.ErrWriter, "CLI Update Check: failed to parse compiled date: %w", err)` every command

Reply:
Ah. I hadn't considered that installing with `go install` would not include the compilation flags from `Makefile`. I'm not sure what we could to to include this case in the other logic.

Either way, I think you're right and it's best to just let this fail silently as opposed to throwing a warning like this constantly.